influxdata/value.go

```diff
@@ -0,0 +1,254 @@
+		// TODO check that it's valid utf-8 data?
+		return Value{
+			bytes: data,
+		}, nil
+	case Unknown:
+		return Value{}, fmt.Errorf("cannot parse value %q with unknown kind", data)
+	default:
+		return Value{}, fmt.Errorf("unexpected value kind %d (value %q)", kind, data)
+	}
+}
+
+// NewValue returns a Value containing the value of x, which must
+// be of type int64 (Int), uint64 (Uint), float64 (Float), bool (Bool),
+// string (String) or []byte (String).
+//
+// Unlike ParseValue, NewValue will make a copy of the byte
+// slice if x is []byte - use ParseValue if you require zero-copy
+// semantics.
+func NewValue(x interface{}) (Value, bool) {
+	switch x := x.(type) {
+	case int64:
+		return Value{
+			number: uint64(x),
+			bytes:  intSentinel[:],
+		}, true
+	case uint64:
+		return Value{
+			number: uint64(x),
+			bytes:  uintSentinel[:],
+		}, true
+	case float64:
+		if math.IsInf(x, 0) || math.IsNaN(x) {
+			return Value{}, false
+		}
+		return Value{
+			number: math.Float64bits(x),
+			bytes:  floatSentinel[:],
+		}, true
+	case bool:
+		n := uint64(0)
+		if x {
+			n = 1
+		}
+		return Value{
+			number: uint64(n),
+			bytes:  boolSentinel[:],
+		}, true
+	case string:
+		return Value{
+			bytes: []byte(x),
+		}, true
+	case []byte:
+		return Value{
+			bytes: append([]byte(nil), x...),
+		}, true
+	}
+	return Value{}, false
+}
+
+// IntV returns the value as an int64. It panics if v.Kind is not Int.
+func (v Value) IntV() int64 {
+	v.mustBe(Int)
+	return int64(v.number)
+}
+
+// UintV returns the value as a uint64. It panics if v.Kind is not Uint.
+func (v Value) UintV() uint64 {
+	v.mustBe(Uint)
+	return v.number
+}
+
+// FloatV returns the value as a float64. It panics if v.Kind is not Float.
+func (v Value) FloatV() float64 {
+	v.mustBe(Float)
+	return math.Float64frombits(v.number)
+}
+
+// StringV returns the value as a string. It panics if v.Kind is not String.
+func (v Value) StringV() string {
+	v.mustBe(String)
+	return string(v.bytes)
+}
+
+// BytesV returns the value as a []byte. It panics if v.Kind is not String.
+// Note that this may return a direct reference to the byte slice within the
+// value - modifying the returned byte slice may mutate the contents
+// of the Value.
+func (v Value) BytesV() []byte {
+	v.mustBe(String)
+	return v.bytes
+}
+
+// BoolV returns the value as a bool. It panics if v.Kind is not Bool.
+func (v Value) BoolV() bool {
+	v.mustBe(Bool)
+	return v.number != 0
+}
+
+// Interface returns the value as an interface. The returned value
+// will have a different dynamic type depending on the value kind;
+// one of int64 (Int), uint64 (Uint), float64 (Float), string (String), bool (Bool).
+func (v Value) Interface() interface{} {
+	switch v.Kind() {
+	case Int:
+		return v.IntV()
+	case Uint:
+		return v.UintV()
+	case String:
+		return v.StringV()
+	case Bool:
+		return v.BoolV()
+	case Float:
+		return v.FloatV()
+	default:
+		// Shouldn't be able to happen.
+		panic("unknown value kind")
```

Comment:
Minor nit but it seems like it would be easier to debug if we printed the value of kind in the panic:

```go
panic(fmt.Errorf("unknown value kind: %v", v.Kind()))
```

Reply:
oops, sorry, forgot to address this before clicking merge. will fix in subsequent PR.